includes/Core/Modules/Module_Settings.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Modules\Module_Settings
+ *
+ * @package   Google\Site_Kit\Core\Modules
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Modules;
+
+use Google\Site_Kit\Core\Storage\Setting;
+
+/**
+ * Base class for module settings.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+abstract class Module_Settings extends Setting {
+
+	/**
+	 * Registers the setting in WordPress.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+		parent::register();
+		$this->add_option_default_filters();
+	}
+
+	/**
+	 * Registers a filter to ensure default values are present in the saved option.
+	 *
+	 * @since n.e.x.t
+	 */
+	protected function add_option_default_filters() {
+		add_filter(
+			'option_' . static::OPTION,
+			function ( $option ) {
+				if ( ! is_array( $option ) ) {
+					return $this->get_default();
+				}
+				return $option;
+			},
+			0
```

Comment:
Should this maybe run on -99 instead, to make sure it's early enough? The legacy key migration runs on 0 as well. It should be later because it's hooked in later, but I think we should "make sure more" that this here really runs first. Or would there be a reason to not run it first?

Reply:
> Should this maybe run on -99 instead, to make sure it's early enough? The legacy key migration runs on 0 as well.

The common priority was intentional. This doesn't conflict with the legacy key migration because the two operate on opposite sides of the `is_array` condition. If it is not an array, it loads the defaults here. It doesn't matter much if legacy key migration runs before or after because it would have no effect since it only runs if the value is an array, but if the current key (set by defaults) is set, it won't do anything.

As for the negative priority, I'm not a big fan of using it. I would opt for using `0` and `1` or `2` perhaps, but that's just a preference (I don't think core uses negative priorities either).

Reply:
That's fair, we don't have to change it due to the `is_array` handling.

There are generally cases though IMO that the negative priority "makes sense" - it's not great, but if you really need to make sure something runs early, I think it's still the most reasonable approach sometimes in WordPress. It's indeed not that crucial here though.